### PR TITLE
Pick up previous Swagger Fix in 2019-09-01 version

### PR DESCRIPTION
**Pull request by jiacheng-L**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/jiacheng-L/azure-rest-api-specs/dev-keyvault-Microsoft.KeyVault-2021-04-01-preview-swaggerFix...master/).

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json [hunk 1]
```diff
@@ -1320,7 +1325,11 @@
           "x-ms-enum": {
             "name": "CreateMode",
             "modelAsString": false
-          }
+          },
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json:1328]: Why was this added? It should have `read` as well (our generators will always construct a readable property), in which case it has all three possible values it would've had already, i.e. this seems redundant.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/providers.json [hunk 2]
```diff
@@ -158,6 +165,81 @@
...
+    "MetricSpecification": {
+      "description": "Metric specification of operation.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of metric specification."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display name of Metric specification."
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/providers.json:178]: Why is "Metric" capitalized here and below when it's not on line 174. It's not a proper noun and shouldn't be capitalized within a sentence.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/providers.json [hunk 3]
```diff
@@ -158,6 +165,81 @@
...
+          "type": "string",
+          "description": "Name of dimension."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display name of dimension."
+        },
+        "toBeExportedForShoebox": {
+          "type": "boolean",
+          "description": "Property to specify whether the dimension should be exported for shoebox."
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/providers.json:240]: Is "shoebox" a product name (first- or third-party)? If so, it should probably be capitalized depending on its marketing rules.

**Review by heaths — APPROVED** [review]

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json (hunk 1 above)
Comment (jiacheng-L) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json:1328]: It was write-only property in the previous version and is caused by swagger correctness issues before: https://github.com/Azure/azure-rest-api-specs/pull/13196/files/bf3d9933cef9afd516920b0a43ccf890f9c7533f#r584399451

**Review by JeffreyRichter — APPROVED** [review]

**Comment by zhenglaizhang** [comment]
@ArcturusZhang looks like the go breaking change is false alarm for this PR, pls help double confirm it.

**Comment by jiacheng-L** [comment]
> @ArcturusZhang looks like the go breaking change is false alarm for this PR, pls help double confirm it.

I believed the GO breaking change is not introduced in this PR, but let's get double confirmation. And we will request GO SDK release to fix that once we fix all swagger errors.

**Comment by ArcturusZhang** [comment]
@zhenglaizhang @jiacheng-L the breaking changes in go are accumulated from previous PRs, we could ignore them here.
